src/huggingface_hub/repository.py

```diff
@@ -11,6 +11,38 @@
 logger = logging.getLogger(__name__)
 
 
+def is_git_repo(folder):
+    """
+    Check if the folder is the root of a git repository
+    """
+    folder_exists = os.path.exists(os.path.join(folder, ".git"))
+    git_branch = subprocess.run(
+        "git branch".split(), cwd=folder, stdout=subprocess.PIPE, stderr=subprocess.PIPE
+    )
+    return folder_exists and git_branch.returncode == 0
+
+
+def is_local_clone(folder, remote_url):
+    """
+    Check if the folder is the a local clone of the remote_url
+    """
+    if not is_git_repo(folder):
+        return False
+
+    remotes = subprocess.run(
+        "git remote -v".split(),
+        stderr=subprocess.PIPE,
+        stdout=subprocess.PIPE,
+        check=True,
+        encoding="utf-8",
+        cwd=folder,
+    ).stdout
+
+    # Remove token for the test with remotes.
+    remote_url = re.sub(r"https://.*@", "https://", remote_url)
+    return remote_url in remotes.split()
```

Comment:
Added those two methods so that they may be used downstream in `transformers` cc @sgugger.

I haven't added them to the `__init__`, so you would use them as such:

```py
from huggingface_hub.repository import is_git_repo, is_local_clone
```

Reply:
You rock!

Reply:
(nit) shouldn't you apply the regex sub to the remotes in `remotes.split()`?